src/util/modules.py

```diff
@@ -882,6 +893,7 @@ def _trigger_auto_discovery(cr):
     # Use accumulated values for the auto_install and force_upgrade modules.
 
     force_installs = ENVIRON["__modules_auto_discovery_force_installs"]
+    _assert_modules_exists(cr, *force_installs)
```

Comment:
Not needed (and done too early).
This is done by the call to `_force_install_module` line 933.
```suggestion
```

Reply:
the condition on line 932 only installs modules that are in the graph and in the force_install list, so we will ignore any unknown modules

Reply:
True.
The call should actually be done after the processing of the graph.
We could also verify the keys of `ENVIRON["__modules_auto_discovery_force_upgrades"]`